cli/azd/cmd/deploy.go

```diff
@@ -207,5 +207,6 @@ func reportServiceDeploymentResultInteractive(svc *project.Service, sdr *project
 		builder.WriteString(fmt.Sprintf(" - Endpoint: %s\n", withLinkFormat(endpoint)))
 	}
 
-	fmt.Println(builder.String())
+	printWithStyling(builder.String())
```

Comment:
Could you help me better understand the terminal you're using (and OS version) here?

This `printWithStyling` stuff seems fragile since it introduces this Windows specific behavior (that I think behaves correctly when using Windows Terminal, even with cmd.exe as the command interpreter) and we keep missing writing with this.

When we introduced it, I wondered if there was a simpler way to solve this by turning on escape code processing:

> I'm fine with this as is - but I do think that longer term we should just eschew all this complexity and instead just set `ENABLE_VIRTUAL_TERMINAL_PROCESSING` if it is unset on the console when azd starts.
>
> Then, we can just always write text with ANSI escape codes to stdout and things will just work.
> 
> `ENABLE_VIRTUAL_TERMINAL_PROCESSING` was added in Windows 10 Anniversary Update (many years ago) and all supported versions of Windows 10 should now support it.
>
> The only gap would be customers on Windows 8.1 (out of standard support now and extended support ends Jan '23) or earlier (totally unsupported).
>
> Honestly, I expect most folks who use windows and azd to be on at least Windows 10 (so they can have WSL) and so I think if we moved to this logic now, we'd be fine - but I'm happy to stick with this for a bit.

I don't want to block this going forward - but I want to understand how to reproduce the issue you saw so I can see if this idea would fix it.

Reply:
@puicchan had reproduced this on cmd prompt shell running in Windows Terminal. I similarly experienced it in both PowerShell and cmd prompt shell running Windows Terminal, on Windows 10.

Reply:
Agreed about this being fragile. I'm leaning towards a solution that ensures we support other Windows 10 and up users, while providing a *reasonable* experience on older Windows versions:

- Enable `ENABLE_VIRTUAL_TERMINAL_PROCESSING` by default. We can check the return code to see if the feature isn't supported for the particular version of Windows.
- If virtual terminal processing isn't supported, we simply set `color.NoColor` to `true`, effectively disabling all color output. That way, users do not see escape characters show up in their console, has a reasonable experience, and we also don't eat the cost of maintaining this largely deprecated path.

Reply:
> I'm leaning towards a solution that ensures we support other Windows 10 and up users, while providing a reasonable experience on older Windows versions:

Sounds Great - 100% in agreement here.

Reply:
Created #215 , will follow-up with those changes soon.